feature-libs/customer-ticketing/components/customer-ticketing-messages/customer-ticketing-messages.component.ts

```diff
@@ -1,112 +1,110 @@
+  ticketDetails$: Observable<TicketDetails | undefined> =
+    this.customerTicketingFacade.getTicket();
 
-  constructor(protected customerTicketingConfig: CustomerTicketingConfig) {}
+  constructor(
+    protected customerTicketingConfig: CustomerTicketingConfig,
+    protected customerTicketingFacade: CustomerTicketingFacade,
+    protected eventService: EventService
+  ) {}
+
+  subscription = new Subscription();
 
   messageEvents$: Observable<Array<MessageEvent> | undefined> =
     this.prepareMessageEvents();
 
   messagingConfigs: MessagingConfigs = this.prepareMessagingConfigs();
 
-  onSend(_event: { files: FileList | undefined; message: string }) {
-    // call to submit new event and upload attachment
+  onSend(event: { files: FileList | undefined; message: string }) {
+    this.subscription.add(
+      this.customerTicketingFacade
+        .createTicketEvent(this.prepareTicketEvent(event.message))
+        .subscribe((createdEvent: TicketEvent) => {
+          if (event.files?.length && createdEvent.code) {
+            this.customerTicketingFacade.uploadAttachment(
+              event.files.item(0),
+              createdEvent.code
+            );
+          } else {
+            this.eventService.dispatch({ status }, TicketEventCreatedEvent);
+          }
+        })
+    );
+  }
+
+  downloadAttachment(event: {
+    messageCode: string;
+    attachmentId: string;
+    fileName: string;
+  }) {
+    this.subscription.add(
+      this.customerTicketingFacade
+        .downloadAttachment(event.messageCode, event.attachmentId)
+        .subscribe((data) => {
+          const downloadURL = window.URL.createObjectURL(data as any);
+          const link = document.createElement('a');
+          link.href = downloadURL;
+          link.download = event.fileName;
+          link.click();
+        })
+    );
   }
 
-  prepareMessageEvents(): Observable<Array<MessageEvent> | undefined> {
+  protected prepareMessageEvents(): Observable<
+    Array<MessageEvent> | undefined
+  > {
     return this.ticketDetails$.pipe(
       map((ticket) =>
-        ticket.ticketEvents?.map(
-          (event): MessageEvent => ({
+        ticket?.ticketEvents?.reverse().map(
+          (event: TicketEvent): MessageEvent => ({
             ...event,
             text: event.message,
-            rightAlign: event.addedByAgent,
+            rightAlign: event.addedByAgent || false,
+            attachments: event.ticketEventAttachments,
           })
         )
       )
     );
   }
 
-  prepareMessagingConfigs(): MessagingConfigs {
+  protected prepareMessagingConfigs(): MessagingConfigs {
     return {
       attachmentRestrictions:
         this.customerTicketingConfig.customerTicketing?.attachmentRestrictions,
       charactersLimit:
         this.customerTicketingConfig.customerTicketing?.inputCharactersLimit,
       enableFileUploadOption: true,
+      displayInput: this.ticketDetails$.pipe(
+        map((ticket) => ticket?.status?.id !== STATUS.CLOSED)
```

Comment:
According to this: "ticket" or its "status" could be undefined and therefore different to CLOSED, which as a consequence will evaluate displayAddMessageSection to TRUE

Reply:
Yeah, but we only want to hide the addMessageSection when the status is closed.